plugins/opener/src/reveal_item_in_dir.rs

```diff
@@ -0,0 +1,192 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: MIT
+
+use std::path::Path;
+
+/// Reveal a path the system's default explorer.
+///
+/// ## Platform-specific:
+///
+/// - **Android / iOS:** Unsupported.
+pub fn reveal_item_in_dir<P: AsRef<Path>>(path: P) -> crate::Result<()> {
+    let path = path.as_ref().canonicalize()?;
+
+    #[cfg(any(
+        windows,
+        target_os = "macos",
+        target_os = "linux",
+        target_os = "dragonfly",
+        target_os = "freebsd",
+        target_os = "netbsd",
+        target_os = "openbsd"
+    ))]
+    return imp::reveal_item_in_dir(&path);
+
+    #[cfg(not(any(
+        windows,
+        target_os = "macos",
+        target_os = "linux",
+        target_os = "dragonfly",
+        target_os = "freebsd",
+        target_os = "netbsd",
+        target_os = "openbsd"
+    )))]
+    Err(crate::Error::UnsupportedPlatform)
+}
+
+#[cfg(windows)]
+mod imp {
+    use super::*;
+
+    use windows::{
+        core::{w, HSTRING, PCWSTR},
+        Win32::{
+            Foundation::ERROR_FILE_NOT_FOUND,
+            System::Com::CoInitialize,
+            UI::{
+                Shell::{
+                    ILCreateFromPathW, ILFree, SHOpenFolderAndSelectItems, ShellExecuteExW,
+                    SHELLEXECUTEINFOW,
+                },
+                WindowsAndMessaging::SW_SHOWNORMAL,
+            },
+        },
+    };
+
+    pub fn reveal_item_in_dir(path: &Path) -> crate::Result<()> {
+        let file = dunce::simplified(path);
+
+        let _ = unsafe { CoInitialize(None) };
+
+        let dir = file
+            .parent()
+            .ok_or_else(|| crate::Error::NoParent(file.to_path_buf()))?;
+
+        let dir = HSTRING::from(dir);
+        let dir_item = unsafe { ILCreateFromPathW(&dir) };
+
+        let file_h = HSTRING::from(file);
+        let file_item = unsafe { ILCreateFromPathW(&file_h) };
+
+        unsafe {
+            if let Err(e) = SHOpenFolderAndSelectItems(dir_item, Some(&[file_item]), 0) {
+                if e.code().0 == ERROR_FILE_NOT_FOUND.0 as i32 {
+                    let is_dir = std::fs::metadata(file).map(|f| f.is_dir()).unwrap_or(false);
+                    let mut info = SHELLEXECUTEINFOW {
+                        cbSize: std::mem::size_of::<SHELLEXECUTEINFOW>() as _,
+                        nShow: SW_SHOWNORMAL.0,
+                        lpVerb: if is_dir {
+                            w!("explore")
+                        } else {
+                            PCWSTR::null()
+                        },
+                        lpClass: if is_dir { w!("folder") } else { PCWSTR::null() },
+                        lpFile: PCWSTR(file_h.as_ptr()),
+                        ..std::mem::zeroed()
+                    };
+
+                    ShellExecuteExW(&mut info).inspect_err(|_| {
+                        ILFree(Some(dir_item));
+                        ILFree(Some(file_item));
+                    })?;
+                }
```

Comment:
After a closer look, I find that this fallback doesn't quite make sense, this opens the file instead of revealing it

Reply:
nice catch, fixed